generators/app/index.js

```diff
@@ -31,121 +32,120 @@ module.exports = yeoman.Base.extend({
-        this.destinationPath('index.js')
-      );
-
-      this.fs.copy(
-        this.templatePath('manifest.js'),
-        this.destinationPath('manifest.js')
-      );
-
-      this.fs.copy(
-        this.templatePath('server.js'),
-        this.destinationPath('server.js')
-      );
-    },
-
-    jsonTemplates: function () {
-      this.directory(
-        this.templatePath('json-templates'),
-        this.destinationPath('json-templates')
-      );
-    },
-
-    serverFiles: function () {
-      this.directory(
-        this.templatePath('server'),
-        this.destinationPath('server')
-      );
-    },
-
-    testFiles: function () {
-      this.directory(
-        this.templatePath('test'),
-        this.destinationPath('test')
-      );
-    }
+    });
+  }
 
-  },
+  writing() {
+  // Dotfiles
+    this.fs.copy(
+      this.templatePath('editorconfig'),
+      this.destinationPath('.editorconfig')
+    );
+    this.fs.copyTpl(
+      this.templatePath('_env'),
+      this.destinationPath('.env'), {
+        serverPort: this.props.serverPort,
+        apiPrefix: this.props.apiPrefix
+      }
+    );
+    this.fs.copy(
+      this.templatePath('eslintrc'),
+      this.destinationPath('.eslintrc')
+    );
+    this.fs.copy(
+      this.templatePath('gitattributes'),
+      this.destinationPath('.gitattributes')
+    );
+    this.fs.copy(
+      this.templatePath('gitignore'),
+      this.destinationPath('.gitignore')
+    );
+
+  // Meta Files
+    this.fs.copy(
+      this.templatePath('LICENSE'),
+      this.destinationPath('LICENSE')
+    );
+    this.fs.copy(
+      this.templatePath('nodemon.json'),
+      this.destinationPath('nodemon.json')
+    );
+    this.fs.copy(
+      this.templatePath('package.json'),
+      this.destinationPath('package.json')
+    );
+    this.fs.copy(
+      this.templatePath('README.md'),
+      this.destinationPath('README.md')
+    );
+    this.fs.copy(
+      this.templatePath('yarn.lock'),
+      this.destinationPath('yarn.lock')
+    );
+
+  // Root JS files
+    this.fs.copy(
+      this.templatePath('config.js'),
+      this.destinationPath('config.js')
+    );
+    this.fs.copy(
+      this.templatePath('index.js'),
+      this.destinationPath('index.js')
+    );
+
+    this.fs.copy(
+      this.templatePath('manifest.js'),
+      this.destinationPath('manifest.js')
+    );
+
+    this.fs.copy(
+      this.templatePath('server.js'),
+      this.destinationPath('server.js')
+    );
+
+  // JSON templates
+    this.fs.copy(
+      this.templatePath('json-templates/gitkeep'),
+      this.destinationPath('json-templates/.gitkeep')
+    );
+
+  // Server files
+    this.fs.copy(
+      this.templatePath('server'),
+      this.destinationPath('server')
+    );
+
+  // Test files
+    this.fs.copy(
+      this.templatePath('test'),
+      this.destinationPath('test')
+    );
+  }
 
-  install: function () {
-    var that = this;
+  install() {
+    const that = this;
```

Comment:
Nope. You should use arrow functions wherever you need `this` inside the `install()` method.

So refactor `callback(error)` to `callback: (error) =>` and you are good to go.